src/zos_files/zowe/zos_files_for_zowe_sdk/files.py

```diff
@@ -146,6 +160,31 @@ def get_dsn_binary_content(self, dataset_name, with_prefixes=False):
         content = self.request_handler.perform_request("GET", custom_args)
         return content
 
+    def get_dsn_binary_content_streamed(self, dataset_name, with_prefixes=False):
+        """
+        Retrieve the contents of a given dataset as a binary bytes object streamed.
+
+        Parameters
+        ----------
+        dataset_name: str - Name of the dataset to retrieve
+        with_prefixes: boolean - if True include a 4 byte big endian record len prefix
+                                 default: False 
+        Returns
+        -------
+        bytes
+            The contents of the dataset with no transformation
+        """
+        custom_args = self.create_custom_request_arguments()
+        custom_args["url"] = "{}ds/{}".format(self.request_endpoint, dataset_name)
+        custom_args["headers"]["Accept"] = "application/octet-stream"
+        if with_prefixes:
+            custom_args["headers"]["X-IBM-Data-Type"] = 'record'
+        else:
+            custom_args["headers"]["X-IBM-Data-Type"] = 'binary'
+        custom_args["stream"] = True
+        content = self.request_handler.perform_request("GET", custom_args)
+        return content
```

Comment:
In addition to setting `stream = True`, the `perform_request` method needs to be changed to use a method that returns a stream instead of a string. See comment above.